src/Numerics/Mesh/Grids.jl

```diff
@@ -678,13 +835,42 @@ function computegeometry(
             if !(dim == 3 && d == 2)
                 ωf = reverse(ωf)
             end
-            sM[1:Nfp[d], f] = dim > 1 ? kron(1, ωf...) : one(FT)
+            sM[1:Nfp[d], f] .= dim > 1 ? kron(1, ωf...) : one(FT)
         end
     end
     sMJ .= sM .* sJ
 
+    # compute MH and JvC
+    horizontal_vertical_metrics(vgeo, Nq, ω)
+
+    # This is mainly done to support FVM plotting when N=0 (since we need cell
+    # edge values)
+    x_vtk = (vgeo[:, _x1, :], vgeo[:, _x2, :], vgeo[:, _x3, :])
+
+    (vgeo, sgeo, x_vtk)
+end
+
+function horizontal_vertical_metrics(vgeo, Nq, ω)
+    dim = length(Nq)
+
+    MH = dim == 1 ? 1 : kron(ones(1, Nq[dim]), reverse(ω[1:(dim - 1)])...)[:]
+    M = kron(1, reverse(ω)...)[:]
+
+    (
+        #! format: off
+        ξ1x1, ξ2x1, ξ3x1, ξ1x2, ξ2x2, ξ3x2, ξ1x3, ξ2x3, ξ3x3,
+        MJ, MJI, MHJH,
+        x1, x2, x3,
+        JcV,
+       #! format: on
+    ) = ntuple(j -> (@view vgeo[:, j, :]), _nvgeo)
+    J = MJ ./ M[:]
```

Comment:
What is MJ, is it updated before in the computegeometry_fvm?

Reply:
`MJ` is the mass matrix with the jacobian determinant scaled in, and `M` is the reference matrix.

(This is a bit different than in the DG kernels where `M` is the same as `MJ`)

It is computed in the FVM geometry computation before this is called (so this is the `N = 0` mass matrix)